scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts

```diff
@@ -0,0 +1,169 @@
+  // SETUP BEGINS HERE
+  //////////////////////////////////////////////////////////////////////////////
+  let tx = null;
+
+  // Allowlist the Minter on the Core contract.
+  tx = await genArt721CoreFlex
+    .connect(deployer)
+    .addMintWhitelisted(genArt721Minter.address);
+
+  await tx.wait();
+
+  console.log(`Allowlisted the Minter Filter on the Core contract.`);
+
+  await delay(EXTRA_DELAY_BETWEEN_TX);
+
+  // Update the Renderer provider.
+  tx = await genArt721CoreFlex
+    .connect(deployer)
+    .updateRenderProviderAddress(artblocksAddress);
+
+  await tx.wait();
+
+  console.log(`Updated the render provider address to: ${artblocksAddress}.`);
+
+  // Set Minter owner.
+  tx = await genArt721Minter.connect(deployer).setOwnerAddress(transferAddress);
+  console.log(`Set the Minter owner to: ${transferAddress}.`);
+  await tx.wait();
+
+  await delay(EXTRA_DELAY_BETWEEN_TX);
+
+  // Allowlist AB staff
+  console.log(`Detected testnet - Adding AB staff to the whitelist.`);
+  const devAddresses = [
+    "0xB8559AF91377e5BaB052A4E9a5088cB65a9a4d63", // purplehat
+    "0x3c3cAb03C83E48e2E773ef5FC86F52aD2B15a5b0", // dogbot
+    "0x0B7917b62BC98967e06e80EFBa9aBcAcCF3d4928", // ben_thank_you
+    "0x3b9038fa89783CBA1933c1689043b4dae2032d1c", // heylinds
+    "0x48631d49245185cB38C01a31E0AFFfA75c133d9a", // meck
+  ];
+  for (let i = 0; i < devAddresses.length; i++) {
+    tx = await genArt721CoreFlex
+      .connect(deployer)
+      .addWhitelisted(devAddresses[i]);
+    await tx.wait();
+
+    console.log(`Allowlisted ${devAddresses[i]} on the Core contract.`);
+    await delay(EXTRA_DELAY_BETWEEN_TX);
+  }
+
+  // (optional) add initial project
+  if (doAddProjectZero) {
+    tx = await genArt721CoreFlex.addProject(
+      projectZeroName,
+      artistAddress,
+      pricePerTokenInWei
+    );
+    await tx.wait();
+  }
+  console.log(
+    `Added project zero ${projectZeroName} on core contract at ${genArt721CoreFlex.address}.`
+  );
+
+  // Allowlist new owner.
+  tx = await genArt721CoreFlex
+    .connect(deployer)
+    .addWhitelisted(transferAddress);
+  await tx.wait();
+
+  console.log(`Allowlisted Core contract access for: ${transferAddress}.`);
+  await delay(EXTRA_DELAY_BETWEEN_TX);
+
+  // Transfer Core contract to new owner.
+  tx = await genArt721CoreFlex.connect(deployer).updateAdmin(transferAddress);
+  await tx.wait();
+  console.log(`Transferred Core contract admin to: ${transferAddress}.`);
+
+  // Output instructions for manual Etherscan verification.
+  const standardVerify = "yarn hardhat verify";
+  console.log(`Verify core contract deployment with:`);
+  console.log(
+    `${standardVerify} --network ${networkName} ${genArt721CoreFlex.address} "${tokenName}" "${tokenTicker}" ${randomizerAddress}`
+  );
+  console.log(`Verify Minter deployment with:`);
+  console.log(
+    `${standardVerify} --network ${networkName} ${genArt721Minter.address} ${genArt721CoreFlex.address}`
+  );
```

Comment:
does verification via etherscan work? or is there a different verification process that we need to go through? If different, I would vote to remove this section from this script.

Reply:
i dont think palm has an etherscan based block explorer, only https://explorer.palm.io/ as far as im aware. i noticed they support something called sourcify, perhaps something that can still be automated (at a later time): https://docs.sourcify.dev/docs/intro

Reply:
It looks like this community hardhat plugin might do the trick! https://www.npmjs.com/package/hardhat-sourcify
@lindsgil - do you mind seeing if that plugin meets our needs here?

Reply:
Thanks for sending that plugin @ryley-o ! I'm getting some weird behavior when I install it.. I'll sidebar you and Ben on slack and see if y'all have any ideas. I was able to "partially" verify the contract through the Sourcify.dev UI which opened up the "Read Contract" and "Write Contract" tabs in the block explorer (check it out here: https://explorer.palm-uat.xyz/address/0x4C7E1e1Ba5e35934C8323d75AF838fDa26588177/contracts#address-tabs) .. Did this by zipping the build-info json file with the sol contract file and submitting here: https://sourcify.dev/#/verifier